python/ml4ir/base/features/feature_fns/categorical.py

```diff
@@ -215,6 +220,107 @@ def categorical_embedding_with_vocabulary_file(feature_tensor, feature_info, fil
     return embedding
 
 
+class CategoricalDropout(layers.Layer):
+    """Custom Dropout class for categorical indices"""
+
+    def __init__(self, dropout_rate, seed=None, **kwargs):
+        """
+        Args:
+            dropout_rate: fraction of units to drop, i.e., set to OOV token 0
+            seed: random seed for sampling
```

Comment:
It's not about sampling here, is it? For sampling what?

Reply:
Sampling to select which values to mask/drop

Reply:
We should add this context :)